src/d/actor/d_a_tbox.cpp

```diff
@@ -314,8 +348,98 @@ void daTbox_c::demoProcAppear() {
 }
 
 /* 00001E4C-0000210C       .text demoProc__8daTbox_cFv */
-void daTbox_c::demoProc() {
-    /* Nonmatching */
+s32 daTbox_c::demoProc() {
+    static char* action_table[] = {
+        "WAIT",
+        "OPEN",
+        "APPEAR",
+        "OPEN_SHORT"
+    };
+
+    s32 actionIdx = dComIfGp_evmng_getMyActIdx(mStaffId, action_table, 4, 0, 0);
+    bool bIsAdvance = dComIfGp_evmng_getIsAddvance(mStaffId);
+
+    if (bIsAdvance) {
+        m03F4 = 0;
+
+        switch (actionIdx) {
+            case DEMO_PROC_OPEN:
+                OpenInit();
+                lightReady();
+                mPLight.mPower = 0.0f;
+                mEfLight.mPower = 0.0f;
+                break;
+            case DEMO_PROC_APPEAR:
+                m03F0 |= 0x20;
+                m03EC = -130.0f;
+                
+                setDzb();
+
+                if (getFuncType() == FUNC_TYPE_TACT) {
+                    m03F0 &= -3;
```

Comment:
if this is a bitfield it might be more readable as `m03F0 &= ~2;` because of two's complement